### ISSUE TITLE
Should folder counts be recursive... ?

### ISSUE THREAD

**Issue by burtonator** [burtonator/polar-bookshelf#1517]
Initially, Polar computed folder counts recursively. For example if we had the following structure:

/compsci 2
/compsci/linux 2

meaning there were 2 documents in /compsci and then 2 in /compsci/linux

... we would compute the counts recursively and the UI would look like:

/compsci 4
/compsci/linux 2

... but I think this is wrong because it's NOT how filesytems on MacOS / Windows behave which is where most people are coming from and what they're used to.

In 2.0 I changed it so that it would ONLY show the folders directly in that folder and wouldn't compute the counts recursively.

I think this is the right strategy since it won't confuse existing users.

**Comment by ekaologik** [comment]
For me an recursive-folder structure like you mentioned "/compsci/linux" is just for structuring the files into folders. I see no needs for putting files into "Main-Folders" like "compsci".. but maybe for other there is

**Comment by sopoforic** [comment]
The counts aren't terribly important, but I'd vote for having them be recursive. Every document in a child folder like `/subject/mathematics/logic` is still related to `/subject/mathematics`, so if I want to get an overview of how much I have related to a topic, I'd expect the counts to be recursive in order to reflect this parent-child relationship.

I would *also* like to be able to select between the document list showing documents in "this folder" or "this folder and all children". The count could correspond to how many documents you would see by clicking the folder, so seeing 0 for some top-level folder would make sense in the former case more than the latter.

**Comment by ekaologik** [comment]
I totally agree with sopoforic. Its also strange, that selecting a parent-folder like "mathematics" dont show the files in an subfolder of mathematics.
